src/store/Authentication/getters.ts

```diff
@@ -0,0 +1,51 @@
+import { AccountTypes } from '@bcrs-shared-components/enums'
+import { RootStateIF } from '@/interfaces'
+
+export default {
+  /** Whether the current account is a premium account. */
+  isPremiumAccount (state, getters, rootState: RootStateIF): boolean {
+    return (rootState.account.currentAccount.accountType === AccountTypes.PREMIUM)
+  },
+
+  /** Whether the user is ServiceBC Staff (which is not the same as Staff). */
+  isSbcStaff (state, getters, rootState: RootStateIF): boolean {
+    return (rootState.account.currentAccount.accountType === AccountTypes.SBC_STAFF)
+  },
+
+  /** The user's Keycloak GUID. */
+  getKeycloakGuid (state, getters, rootState, rootGetters): string {
+    return rootGetters['auth/keycloakGuid']
+  },
+
+  /** The user's Keycloak Bearer Token. */
+  getKeycloakToken (state, getters, rootState: RootStateIF): string {
+    return rootState.auth.token
+  },
+
+  /** Returns true is the user is Keycloak authenticated. */
+  isKeycloakAuthenticated (state, getters, rootState, rootGetters): boolean {
+    return rootGetters['auth/isAuthenticated']
+  },
+
+  /** Returns the Keycloak user object. */
+  getKeycloakUser (state, getters, rootState: RootStateIF): any {
+    return rootState.account.currentUser
+  },
```

Comment:
Given that this file contains getters for auth and account, would there be a better name for the current folder?

Reply:
I struggle with the best names for things!  I chose 'Authentication' because I see it as encompassing both the Keycloak login and the selected account.  If you have an alternative suggestion, please let me know.

Reply:
`auth-module` ?

Reply:
Is 'auth' short for 'authentication'?  I thought about adding "module" to each but then I was thinking that gets repetitive.  You make the call.  I'm happy to use whatever you choose.
- store
  - auth-module
  - current-account-module
  - todo-module
  - filing-history-module

Reply:
What is this folder exactly -- store stuff that isn't in our own store?

src/store/getters.ts - ours
src/store/other/getters.ts - external?
src/store/external/getters.ts - external?
src/store/imported/getters.ts - external?

🤷‍♂️

Reply:
This folder is contains our Vuex authentication module.  At present, it contains only getters that reference the rootState or rootGetters for the current Keycloak user or the user's selected account.  In the near future, I'd like to an 'Authorization' module that would contain the code associated with LaunchDarkly as well as the code associated with user authorizations per businessId.  One of the reasons I'd recommend against the name "auth-module" is that "auth" could mean authentication or authorization.

Reply:
We also get authorization info from the auth endpoint.

How about naming it Root and just adding all external store things in this folder?

Reply:
I was hoping to name our store modules by the type of data they grant access to.  For example, the configuration module has configuration data, the filing history module has filing history data, the todo module has ... you get the idea. 

If we name this module "Root" it doesn't tell developers what type of data is contained within this module. Also, consider that this module, may in the future have it's own state, actions and mutators.  For example, we may want to extend the this module to have custom business logic that combines rootState with our own state.  If we name this module "Root" and agree that this module will only contain references to either rootState or rootGetters, where would our own authentication business logic reside?

Reply:
> I was hoping to name our store modules by the type of data they grant access to. For example, the configuration module has configuration data, the filing history module has filing history data, the todo module has ... you get the idea.

Absolutely.

OK, call it Authentication. We can always change it in the future as we get more context/complicated.